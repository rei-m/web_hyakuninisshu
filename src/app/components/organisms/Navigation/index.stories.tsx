import { Meta, StoryObj } from '@storybook/react';
import { Navigation } from './index';

export default {
  title: 'organisms/Navigation',
  component: Navigation,
  decorators: [(story) => <div style={{ width: '500px' }}>{story()}</div>],
} as Meta<typeof Navigation>;

type Story = StoryObj<typeof Navigation>;

export const Default: Story = {
  args: {},
};
