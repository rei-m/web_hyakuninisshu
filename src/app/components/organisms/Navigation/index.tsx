import type { MenuType } from '@/app/components/molecules/NavigationItem';
import type { SxAppProps } from '@/theme';

import Link from 'next/link';
import Box from '@mui/material/Box';
import NavigationItem from '@/app/components/molecules/NavigationItem';

import { WIDTH_SIDE_NAV } from '@/theme';
import { ROUTING } from '@/configs/routing';

export type NavigationProps = {
  currentMenuType?: MenuType;
  sx?: SxAppProps;
};

const LINK_SX: SxAppProps = {
  flexGrow: {
    xs: 1,
    sm: 0,
  },
  width: {
    xs: '100%',
    sm: WIDTH_SIDE_NAV,
  },
};

const Navigation = ({ currentMenuType, sx }: NavigationProps) => (
  <Box
    component={'nav'}
    sx={[
      {
        display: 'flex',
        backgroundColor: 'primary.main',
        boxSizing: 'border-box',
        boxShadow: 1,
      },
      ...(Array.isArray(sx) ? sx : [sx]),
    ]}
  >
    <Box component={Link} href={ROUTING.training()} sx={LINK_SX}>
      <NavigationItem menuType={'training'} active={currentMenuType === 'training'} />
    </Box>
    <Box component={Link} href={ROUTING.exam()} sx={LINK_SX}>
      <NavigationItem menuType={'exam'} active={currentMenuType === 'exam'} />
    </Box>
    <Box component={Link} href={ROUTING.karutas()} sx={LINK_SX}>
      <NavigationItem menuType={'material'} active={currentMenuType === 'material'} />
    </Box>
    <Box component={Link} href={ROUTING.about()} sx={LINK_SX}>
      <NavigationItem menuType={'other'} active={currentMenuType === 'other'} />
    </Box>
  </Box>
);

export default Navigation;
