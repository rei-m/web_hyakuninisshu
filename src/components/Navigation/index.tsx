import * as React from 'react';
import styled from 'styled-components';
import { withAppTheme } from '../../styles';
import { Link, LinkProps } from 'react-router-dom';
import NavIcon from '../NavIcon';
import { withRipple } from '../../enhancers/withRipple';
import { MenuType } from '../../enums';
import { ROUTE_PATHS } from '../../constants';

export interface NavigationProps {
  readonly currentMenuType?: MenuType;
}

const Root = withAppTheme(styled.nav)`
  display: flex;
  position: fixed;
  height: ${({ theme }) => theme.bottomNavHeight};
  width: 100vw;
  bottom: 0;
  background-color: ${({ theme }) => theme.colorPrimary};
  box-shadow: 0 -1px 5px rgba(0, 0, 0, 0.26);
  box-sizing: border-box;

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    height: 100vh;
    width: ${({ theme }) => theme.sideNavWidth};
    padding-top: ${({ theme }) => theme.headerHeightWide};
    flex-direction: column;
  }
`;

const IconBox = styled.span`
  flex-grow: 1;
  width: 100%;
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    flex-grow: 0;
    height: ${({ theme }) => theme.sideNavWidth};
  }
`;

const LinkWithRipple = withRipple<LinkProps>(Link);

const Navigation: React.SFC<NavigationProps> = ({ currentMenuType }) => (
  <Root>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.TRAINING}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Training}
          text="練習"
          isCurrent={currentMenuType === MenuType.Training}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.EXAM}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Exam}
          text="腕試し"
          isCurrent={currentMenuType === MenuType.Exam}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.KARUTAS}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Material}
          text="資料"
          isCurrent={currentMenuType === MenuType.Material}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.ABOUT}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Other}
          text="その他"
          isCurrent={currentMenuType === MenuType.Other}
        />
      </LinkWithRipple>
    </IconBox>
  </Root>
);

export default Navigation;
