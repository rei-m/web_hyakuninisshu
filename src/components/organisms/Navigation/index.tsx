import * as React from 'react';
import { GatsbyLinkProps, Link } from 'gatsby';
import styled from '@src/styles/styled-components';
import NavigationItem from '@src/components/molecules/NavigationItem';
import { menuIcon } from '@src/components/atoms/MenuIcon';
import { MenuType } from '@src/enums';
import { ROUTE_PATHS } from '@src/constants';
import { withRipple } from '@src/enhancers/withRipple';

export interface Props {
  currentMenuType?: MenuType;
  className?: string;
}

const Container = styled.nav`
  display: flex;
  background-color: ${({ theme }) => theme.colorPrimary};
  box-shadow: ${({ theme }) => theme.elevationShadow1x};
  box-sizing: border-box;
`;

// withRippleの作りがイマイチなせいで必要。。。
const ItemWrapper = styled.span`
  flex-grow: 1;
  width: 100%;
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    flex-grow: 0;
    height: ${({ theme }) => theme.sideNavWidth};
  }
`;

const LinkWithRipple = withRipple<GatsbyLinkProps<{}>>(Link);

const renderIcon = (menuType: MenuType) => {
  const Icon = menuIcon(menuType);
  return <Icon />;
};

const WrappedItem = ({ to, menuType, active }: { to: string; menuType: MenuType; active: boolean }) => (
  <ItemWrapper>
    <LinkWithRipple to={to} style={{ width: '100%', height: '100%' }}>
      <NavigationItem menuType={menuType} active={active} renderIcon={renderIcon} />
    </LinkWithRipple>
  </ItemWrapper>
);

const Navigation: React.FC<Props> = ({ currentMenuType, className }) => (
  <Container className={className}>
    <WrappedItem
      to={ROUTE_PATHS.TRAINING}
      menuType={MenuType.Training}
      active={currentMenuType === MenuType.Training}
    />
    <WrappedItem to={ROUTE_PATHS.EXAM} menuType={MenuType.Exam} active={currentMenuType === MenuType.Exam} />
    <WrappedItem to={ROUTE_PATHS.KARUTAS} menuType={MenuType.Material} active={currentMenuType === MenuType.Material} />
    <WrappedItem to={ROUTE_PATHS.ABOUT} menuType={MenuType.Other} active={currentMenuType === MenuType.Other} />
  </Container>
);

export default Navigation;
