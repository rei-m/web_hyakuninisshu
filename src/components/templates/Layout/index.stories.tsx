import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { appContextDecorator } from '@helper/storybook';
import Layout, { Props } from './index';

const props: Props = {
  title: 'れいあうと',
  isDisplayNav: false,
};

storiesOf('templates/Layout', module)
  .addDecorator(story => appContextDecorator(story))
  .add('default', () => <Layout {...props} />)
  .add('with navigation', () => <Layout {...props} isDisplayNav={true} />)
  .add('with back button', () => <Layout {...props} isDisplayNav={true} onClickBack={action('onClickBack')} />)
  .add('with search button', () => <Layout {...props} isDisplayNav={true} onClickSearch={action('onClickSearch')} />);
