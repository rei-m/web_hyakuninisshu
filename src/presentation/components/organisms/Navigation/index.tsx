import React from 'react';
import { GatsbyLinkProps, Link } from 'gatsby';
import clsx from 'clsx';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { paths } from '@src/presentation/routes';
import NavigationItem, { MenuType } from '@src/presentation/components/molecules/NavigationItem';
import { ExamIcon, MaterialIcon, TrainingIcon, OtherIcon } from '@src/presentation/components/atoms/MenuIcon';
import { withRipple } from '@src/presentation/enhancers/withRipple';
import { ThemeInterface } from '@src/presentation/styles/theme';

export type Props = {
  currentMenuType?: MenuType;
  className?: string;
};

const useStyles = makeStyles<ThemeInterface>(theme => ({
  root: {
    display: 'flex',
    backgroundColor: theme.palette.primary.main,
    boxSizing: 'border-box',
    boxShadow: theme.elevationShadow1x,
  },
  itemWrapper: {
    flexGrow: 1,
    width: '100%',
    [`@media screen and (min-width:${theme.minWidthWide})`]: {
      flexGrow: 0,
      height: theme.sideNavWidth,
    },
  },
}));

const LinkWithRipple = withRipple<GatsbyLinkProps<{}>>(Link);

const ICON_MAP = {
  training: TrainingIcon,
  exam: ExamIcon,
  material: MaterialIcon,
  other: OtherIcon,
} as const;

const renderIcon = (menuType: MenuType) => {
  const Icon = ICON_MAP[menuType];
  return <Icon style={{ fontSize: '2.4rem' }} />;
};

const WrappedItem = ({
  to,
  menuType,
  active,
  className,
}: {
  to: string;
  menuType: MenuType;
  active: boolean;
  className: string;
}) => (
  <span className={className}>
    <LinkWithRipple to={to} style={{ width: '100%', height: '100%' }}>
      <NavigationItem menuType={menuType} active={active} renderIcon={renderIcon} />
    </LinkWithRipple>
  </span>
);

const Navigation: React.FC<Props> = ({ currentMenuType, className = '' }) => {
  const classes = useStyles();
  return (
    <nav className={clsx(classes.root, className)}>
      <WrappedItem
        to={paths.training()}
        menuType={'training'}
        active={currentMenuType === 'training'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.exam()}
        menuType={'exam'}
        active={currentMenuType === 'exam'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.karutas()}
        menuType={'material'}
        active={currentMenuType === 'material'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.about()}
        menuType={'other'}
        active={currentMenuType === 'other'}
        className={classes.itemWrapper}
      />
    </nav>
  );
};

export default Navigation;
