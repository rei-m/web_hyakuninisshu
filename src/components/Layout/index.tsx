import * as React from 'react';
import styled from '@src/styles/styled-components';
import Header from '@src/components/Header';
import Navigation from '@src/components/Navigation';
import { MenuType } from '@src/enums';

export interface Props {
  title: string;
  currentMenuType?: MenuType;
  isDisplayNav: boolean;
  onClickBack?: () => void;
  onClickSearch?: () => void;
}

type BodyProps = Pick<Props, 'isDisplayNav'>;

export const Body = styled.div<BodyProps>`
  padding-top: ${({ theme }) => theme.headerHeight};
  padding-bottom: ${({ isDisplayNav, theme }) => (isDisplayNav ? theme.bottomNavHeight : '0')};
  text-align: center;
  min-height: 100vh;

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    padding-top: ${({ theme }) => theme.headerHeightWide};
    padding-bottom: 0;
    padding-left: ${({ isDisplayNav, theme }) => (isDisplayNav ? theme.bottomNavHeight : '0')};
  }
`;

const Layout: React.FC<Props> = ({ title, children, onClickBack, onClickSearch, isDisplayNav, currentMenuType }) => (
  <>
    <Header title={title} onClickBack={onClickBack} onClickSearch={onClickSearch} />
    <Body isDisplayNav={isDisplayNav}>{children}</Body>
    {isDisplayNav && <Navigation currentMenuType={currentMenuType} />}
  </>
);

export default Layout;
