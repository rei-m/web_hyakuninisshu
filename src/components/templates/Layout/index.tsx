import * as React from 'react';
import styled from '@src/styles/styled-components';
import Header from '@src/components/organisms/Header';
import Navigation from '@src/components/organisms/Navigation';
import { MenuType } from '@src/enums';

export interface Props {
  title: string;
  currentMenuType?: MenuType;
  isDisplayNav: boolean;
  onClickBack?: () => void;
  onClickSearch?: () => void;
}

type BodyProps = Pick<Props, 'isDisplayNav'>;

export const Body = styled.main<BodyProps>`
  padding-top: ${({ theme }) => theme.headerHeight};
  padding-bottom: ${({ isDisplayNav, theme }) => (isDisplayNav ? theme.bottomNavHeight : '0')};
  text-align: center;
  min-height: 100vh;

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    padding-top: ${({ theme }) => theme.headerHeightWide};
    padding-bottom: 0;
    padding-left: ${({ isDisplayNav, theme }) => (isDisplayNav ? theme.bottomNavHeight : '0')};
  }
`;

const StyledHeader = styled(Header)`
  left: 0;
  right: 0;
  top: 0;
  z-index: 1;
  position: fixed;
`;

const StyledNavigation = styled(Navigation)`
  position: fixed;
  bottom: 0;
  display: flex;
  flex-direction: row;
  height: ${({ theme }) => theme.bottomNavHeight};
  width: 100vw;
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    flex-direction: column;
    height: 100vh;
    width: ${({ theme }) => theme.sideNavWidth};
    padding-top: ${({ theme }) => theme.headerHeightWide};
  }
`;

const Layout: React.FC<Props> = ({ title, children, onClickBack, onClickSearch, isDisplayNav, currentMenuType }) => (
  <>
    <StyledHeader title={title} onClickBack={onClickBack} onClickSearch={onClickSearch} />
    <Body isDisplayNav={isDisplayNav}>{children}</Body>
    {isDisplayNav && <StyledNavigation currentMenuType={currentMenuType} />}
  </>
);

export default Layout;
