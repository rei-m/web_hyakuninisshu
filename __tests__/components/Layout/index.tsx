import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { shallow } from 'enzyme';
import Layout, { Props } from '@src/components/templates/Layout';
import Navigation from '@src/components/molecules/Navigation';

describe('<Layout />', () => {
  let baseProps: Props;

  beforeEach(() => {
    baseProps = {
      title: 'title',
      isDisplayNav: true,
      onClickBack: jest.fn(),
    };
  });

  describe('with Navigation', () => {
    let props: Props;

    beforeEach(() => {
      props = { ...baseProps };
    });

    it('should render component', () => {
      const renderer = ReactTestRenderer.create(<Layout {...props}>test</Layout>);
      expect(renderer.toJSON()).toMatchSnapshot();
    });

    it('contains Navigation', () => {
      const wrapper = shallow(<Layout {...props}>test</Layout>);
      expect(wrapper.find(Navigation).length).toEqual(1);
    });
  });

  describe('without Navigation', () => {
    let props: Props;

    beforeEach(() => {
      props = { ...baseProps, isDisplayNav: false };
    });

    it('should render component', () => {
      const renderer = ReactTestRenderer.create(<Layout {...props}>test</Layout>);
      expect(renderer.toJSON()).toMatchSnapshot();
    });

    it('not contains Navigation', () => {
      const wrapper = shallow(<Layout {...props}>test</Layout>);
      expect(wrapper.find(Navigation).length).toEqual(0);
    });
  });
});
