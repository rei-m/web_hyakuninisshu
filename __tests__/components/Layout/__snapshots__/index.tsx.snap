// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> with Navigation should render component 1`] = `
Array [
  <header
    className="sc-bwzfXH geYBAS"
  >
    <div
      className="sc-bdVaJa flBAph"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <span
        className="sc-bxivhb iUaUQC"
        data-test="back"
        onClick={[MockFunction]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
    </div>
    <h1
      className="sc-htpNat eSnwWs"
    >
      title
    </h1>
  </header>,
  <div
    className="sc-gzVnrw hzJxWQ"
  >
    test
  </div>,
  <nav
    className="sc-EHOje extRj"
  >
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="/training"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX xOgRI"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="/exam"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX hcwVar"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M22 10l-6-6H4c-1.1 0-2 .9-2 2v12.01c0 1.1.9 1.99 2 1.99l16-.01c1.1 0 2-.89 2-1.99v-8zm-7-4.5l5.5 5.5H15V5.5z"
              />
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="/karutas"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX fGzQGR"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6zm16-4H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-1 9H9V9h10v2zm-4 4H9v-2h6v2zm4-8H9V5h10v2z"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa flBAph"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          href="/about"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <span
            className="sc-ifAKCX btkJVi"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1"
              focusable="false"
              role="presentation"
              style={
                Object {
                  "fontSize": undefined,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h20v20H0V0z"
                fill="none"
                transform="scale(1.2, 1.2)"
              />
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
          </span>
        </a>
      </div>
    </span>
  </nav>,
]
`;

exports[`<Layout /> without Navigation should render component 1`] = `
Array [
  <header
    className="sc-bwzfXH geYBAS"
  >
    <div
      className="sc-bdVaJa flBAph"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <span
        className="sc-bxivhb iUaUQC"
        data-test="back"
        onClick={[MockFunction]}
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </span>
    </div>
    <h1
      className="sc-htpNat eSnwWs"
    >
      title
    </h1>
  </header>,
  <div
    className="sc-gzVnrw eeAPtW"
  >
    test
  </div>,
]
`;
